cpp/src/quantiles/quantiles.hpp

```diff
@@ -19,11 +19,15 @@
 //Quantile (percentile) functionality
 
 #include <cudf/cudf.h>
+#include <cudf/quantiles.hpp>
+#include <utilities/cudf_utils.h>
+#include <utilities/release_assert.cuh>
 
 namespace cudf {
 namespace interpolate {
 
 template <typename T_out, typename T_in>
+CUDA_HOST_DEVICE_CALLABLE
```

Comment:
Is it ever necessary to call this from host code? If not, avoid declaring it __host__ __device__, just make it __device__.

Reply:
This function, and all others in this header, are shared by the series quantile functionality. The series version does it all on host because it uses only two values: first and last.